File: app/src/main/java/com/mnj/java_solutions/gfg/ConvertAnArrayReducedForm.java
package com.mnj.java_solutions.gfg;

import java.util.Arrays;
import java.util.HashMap;

public class ConvertAnArrayReducedForm {

    public void convert() {
        int[] arr = {10, 40, 20};

        HashMap<Integer, Integer> map = new HashMap<>();

        int[] temp = new int[arr.length];

        //Create a copy array
        for (int i = 0; i < arr.length; i++) {
            temp[i] = arr[i];
        }
//sort the array
        Arrays.sort(temp);

        //Create a map (key will array element, and value will be index)
        for (int i = 0; i < temp.length; i++)
            map.put(temp[i], i);

        //Query every given element and find its index in sorted array.

        for (int i = 0; i < arr.length; i++) {
            arr[i] = map.get(arr[i]);
        }

        for (int num : arr) {
            System.out.println("==>> New Array is:" + num);
        }

    }
}

File: app/src/main/java/com/mnj/java_solutions/gfg/ConvertAnArrayReducedForm.kt
package com.mnj.java_solutions.gfg

import java.util.*

class ConvertAnArrayReducedForm {
    fun convert() {
        val arr = intArrayOf(10, 40, 20)
        val map = HashMap<Int, Int>()
        val temp = IntArray(arr.size)

        //Create a copy array
        for (i in arr.indices) {
            temp[i] = arr[i]
        }
        //sort the array
        Arrays.sort(temp)

        //Create a map (key will array element, and value will be index)
        for (i in temp.indices) map[temp[i]] = i

        //Query every given element and find its index in sorted array.
        for (i in arr.indices) {
            arr[i] = map[arr[i]]!!
        }
        for (num in arr) {
            println("==>> New Array is:$num")
        }
    }
}